BREAKING: 13.0 Docs (#5315)
cypress-io/cypress-documentation: docs/guides/references/migration-guide.mdx

title: Migration Guide
---

## Migrating to Cypress 13.0

This guide details the changes and how to change your code to migrate to Cypress
version 13.0.
[See the full changelog for version 12.0](/guides/references/changelog#12-0-0).

### `cy.readFile()` is now a query command

> In Cypress 13, the [`.readFile()`](/api/commands/as) command is now a query.

Reviewer: ```suggestion
In Cypress 13, the [`.readFile()`](/api/commands/readfile) command is now a query.
```

title: Migration Guide
---

## Migrating to Cypress 13.0

This guide details the changes and how to change your code to migrate to Cypress

> version 13.0.

Reviewer 2: or `13.0.0` or `13.x.x`?
```suggestion
version 13.
```

[See the full changelog for version v13.0](/guides/references/changelog#13-0-0).

### Cypress Cloud Test Replay

> [Test Replay](/guides/cloud/debugging/test-replay) is enabled by default in `v13.0` of the Cypress App.

Reviewer 2: for consistency?
```suggestion
[Test Replay](/guides/cloud/debugging/test-replay) is enabled by default in `v13` of the Cypress App.
```

You may need to allowlist `capture.cypress.io` if you work with a strict VPN. See our FAQ section about [VPN subdomain allowlisting](/faq/questions/cloud-faq#Im-working-with-a-restrictive-VPN-Which-subdomains-do-I-have-to-allow-on-my-VPN-for-Cypress-Cloud-to-work-properly).

With Test Replay enabled, the Cypress Runner UI is hidden by default when recording a run to the Cloud. If the Runner UI is needed during the run, you can enable it by passing [`--runner-ui`](/guides/guides/command-line#cypress-run-runner-ui) to the [`cypress run`](/guides/guides/command-line#cypress-run) command.

> You can [opt out](/guides/cloud/debugging/test-replay#Opt-out-of-Test-Replay) of this feature in Cloud project-level settings.

Reviewer 2: ```suggestion
You can [opt-out](/guides/cloud/debugging/test-replay#Opt-out-of-Test-Replay) of this feature in Cloud project-level settings.
```

### Video updates

#### `video` is set to `false` by default

> You can continue recording video, by passing `video: true`, if you want video locally or you want video for some other reason, like in non-Chromium browsers where [Test Replay](/guides/cloud/debugging/test-replay) is not available.

Reviewer 2: ```suggestion
You can continue recording video by setting `video` to `true` either in your Cypress configuration or via [overriding options](/guides/references/configuration#Overriding-Options). This can be useful if you want video locally or want video for some other reason, like in non-Chromium browsers where [Test Replay](/guides/cloud/debugging/test-replay) is not available.
```

If you want to prevent a passing test from uploading to the Cloud, we recommend deleting the video using our [guide with code examples to discard captured video of passing tests](/guides/guides/screenshots-and-videos#Control-which-videos-to-keep-and-upload-to-Cypress-Cloud).

#### `videoCompression` is set to `false` by default

Cypress has the capability to compress recorded videos after a run to reduce the video file size. By default, compression is now turned off. This results in a reduced run time by removing the time to compress the video, a larger video file size and better video quality.

You can enable this with the `videoCompression` [configuration](/guides/references/configuration#Videos) option if you'd like to reduce the video file size for any reason. This will also reduce the video quality and take slightly longer to process and complete the run.

:::cypress-config-example

```ts
{
  // value can be true/false -or- an integer between 0 and 51
  videoCompression: true,
}
```

:::

### `cy.readFile()` is now a query command

> In Cypress 13, the [`.readFile()`](/api/commands/readFile) command is now a query.

Reviewer 2: ```suggestion
In Cypress `v13`, the [`.readFile()`](/api/commands/readFile) command is now a query.
```

Tests written using it should continue to operate exactly as before; no changes
are necessary.

`readFile()` will re-read the file from disk if any upcoming command in the same
chain fails. Assertions no longer have to be directly attached.

```js
cy.readFile(`users.json`).its('users.123.fullName').should('eq', 'John Doe')
```

> Beginning with Cypress 13, the above test will re-read the file until the file

Reviewer 2: ```suggestion
Beginning with Cypress `v13`, the above test will re-read the file until the file
```

title: Migration Guide
---

## Migrating to Cypress 13.0

This guide details the changes and how to change your code to migrate to Cypress
version 13.0.
[See the full changelog for version 13.0](https://docs.cypress.io/guides/overview/why-cypress).

### Cypress Cloud Test Replay

[Test Replay](https://docs.cypress.io/guides/overview/why-cypress) is enabled by default in `v13.0` of the Cypress App.

You may need to allowlist `capture.cypress.io` if you work with a strict VPN. See our FAQ section about [VPN subdomain allowlisting](/faq/questions/cloud-faq#Im-working-with-a-restrictive-VPN-Which-subdomains-do-I-have-to-allow-on-my-VPN-for-Cypress-Cloud-to-work-properly).

You can [opt out](https://docs.cypress.io/guides/overview/why-cypress) of this feature in Cloud project-level settings.

<!---
TODO: replace urls for merge to v13 branch:

> changelog- /guides/references/changelog#13-0-0

Reviewer 2: want to make sure we take care of this before we merge/release.
